packages/legend-application/src/components/shared/PackageableElementOptionRenderer.tsx

```diff
@@ -14,23 +14,64 @@
  * limitations under the License.
  */
 
-import type { PackageableElement } from '@finos/legend-graph';
+import type {
+  PackageableElement,
+  GraphManagerState,
+} from '@finos/legend-graph';
 import type { PackageableElementOption } from '../../stores/shared/PackageableElementOption.js';
 
-export const getPackageableElementOptionalFormatter = (props?: {
+export const getPackageableElementOptionFormatter = (props: {
   darkMode?: boolean;
+  graphManagerState: GraphManagerState;
 }): ((
   option: PackageableElementOption<PackageableElement>,
 ) => React.ReactNode) =>
   function PackageableElementOptionLabel(
     option: PackageableElementOption<PackageableElement>,
   ): React.ReactNode {
-    const className = props?.darkMode
+    const className = props.darkMode
       ? 'packageable-element-format-option-label--dark'
       : 'packageable-element-format-option-label';
 
+    const getColor = (element: PackageableElement): string => {
+      switch (true) {
```

Comment:
this is weird way to use `switch` expression. let's do

```ts
const rootPackageName = getRootPackageName(element);

switch(rootPackageName) {
  case ROOT_PACKAGE_NAME.SYSTEM: { ... }
  ...
}
```

Reply:
👀